kernel/src/process.rs

```diff
@@ -1242,10 +1253,39 @@ impl<C: Chip> ProcessType for Process<'a, C> {
             restart_count,
         ));
 
-        let _ = match last_syscall {
-            Some(syscall) => writer.write_fmt(format_args!(" Last Syscall: {:?}", syscall)),
-            None => writer.write_str(" Last Syscall: None"),
-        };
+        // Print a list of the app's most recently called syscalls and the value
+        // the kernel returned. Should look like:
+        //
+        // ```
+        //  Last Syscalls (most recent first):
+        //     0. YIELD -> None
+        //     1. COMMAND { driver_number: 3, subdriver_number: 1, arg0: 0, arg1: 0 } -> 0
+        //     2. SUBSCRIBE { driver_number: 3, subdriver_number: 0, callback_ptr: 0x300a1, appdata: 0 } -> 0
+        //     3. YIELD -> None
+        //     4. COMMAND { driver_number: 1, subdriver_number: 1, arg0: 5, arg1: 0 } -> 0
+        // ```
+        match last_syscalls {
+            Some(syscalls) => {
+                let _ = writer.write_fmt(format_args!(" Last Syscalls (most recent first):\n"));
+                for i in 0..5 {
+                    match syscalls[i].0 {
+                        Some(syscall) => {
+                            let _ = writer.write_fmt(format_args!("   • {:?} -> ", syscall));
```

Comment:
The bullet point is not the same as advertised in the documentation. And although Tock should support UTF-8, I'm opposing using "non-standard" fancy characters for debugging purposes, we should keep it simple.

Reply:
Why?

Reply:
Unicode characters are hard to type on a keyboard, require a font that can render all of the characters (the list of which changes every month), the non-ASCII characters use more bytes (and Tock doesn't have a lot of margin in terms of memory limits), breaking in the middle of a UTF-8 code point makes the remaining bytes of the code point un-renderable (and Tock does break at arbitrary bytes, see #1449), it can lead to compatibility issues if used with console software that isn't configured for UTF-8, etc.

Because unicode exists doesn't mean we should use it. In this example, indentation (and maybe a simple character such as `-`) is sufficient.